gym-kinova-gripper/check_coordinates_for_collision.py

```diff
@@ -0,0 +1,154 @@
+    env = gym.make('gym_kinova_gripper:kinovagripper-v0')
+    
+    # coords filename structure: "gym_kinova_gripper/envs/kinova_description/obj_hand_coords/" + noise_file + str(mode)+"_coords/" + str(env.orientation) + "/" + random_shape + ".txt"
+    with_grasp = False
+    if with_noise is False:
+        noise_str = "no_noise/"
+    else:
+        noise_str = "with_noise/"
+    mode = "shape"
+    bad_coordinate = False
+
+    # Make the new VALID valid coordinates filepath
+    valid_all_coords_file = "./gym_kinova_gripper/envs/kinova_description/valid_obj_hand_coords_new/"
+    env.create_paths([valid_all_coords_file, valid_all_coords_file + noise_str,
+                       valid_all_coords_file + noise_str + str(mode) + "_coords/",
+                       valid_all_coords_file + noise_str + str(mode) + "_coords/" + str(env.orientation)])
+    valid_all_coords_filepath = valid_all_coords_file + noise_str + str(mode) + "_coords/" + str(env.orientation)
+
+    # BAD coordinates filepath
+    bad_all_coords_file = "./gym_kinova_gripper/envs/kinova_description/bad_obj_hand_coords_new/"
+    env.create_paths([bad_all_coords_file, bad_all_coords_file + noise_str,
+                       bad_all_coords_file + noise_str + str(mode) + "_coords/",
+                       bad_all_coords_file + noise_str + str(mode) + "_coords/" + str(env.orientation)])
+    bad_all_coords_filepath = bad_all_coords_file + noise_str + str(mode) + "_coords/" + str(env.orientation)
+
+    for shape_name in shape_keys:
+        valid_data = []
+        bad_data =[]
+        valid_shape_coords_file = valid_all_coords_filepath + "/" + shape_name + ".txt"
+        bad_shape_coords_file = bad_all_coords_filepath + "/" + shape_name + ".txt"
+        requested_shapes = [shape_name]
+
+        # Generate randomized list of objects to select from
+        env.Generate_Latin_Square(file_size, "objects.csv", shape_keys=requested_shapes)
+
+        # valid coordinates for plotting
+        local_valid_x = []
+        local_valid_y = []
+        global_valid_x = []
+        global_valid_y = []
+
+        # Bad coordinates for plotting
+        local_bad_x = []
+        local_bad_y = []
+        global_bad_x = []
+        global_bad_y = []
+
+        # Check each coordinate within the file
+        for orient_idx in range(file_size):
+            print("Coord File Idx: ", orient_idx)
+            state = env.reset(shape_keys=requested_shapes, hand_orientation=hand_orientation, with_grasp=with_grasp,
+                               env_name="env", mode=mode, orient_idx=orient_idx, with_noise=with_noise)
+
+            # Print which file you're running through
+            if orient_idx == 0:
+                coords_file = env.get_coords_filename()
+                print("Coords filename: ", coords_file)
+
+            # Get the current object coordinate and hand orientation pair
+            obj_coords = env.get_obj_coords()
```

Comment:
Needs to be updated: previous object coordinates are not being updated. Need to update this in order to correctly check for collision.